plasmapy/physics/magnetostatics.py

```diff
@@ -0,0 +1,326 @@
+"""
+Define MagneticStatics class to calculate common static magnetic fields(issue #100).
+"""
+
+import abc
+from astropy import units as u, constants
+import numpy as np
+
+from typing import (
+    Callable
+)
+
+from scipy.special import roots_legendre
+
+DEFAULT_GENERAL_INTEGRATION_POINTS = 1000
+DEFAULT_CIRCULAR_INTEGRATION_POINTS = 300
+
+
+class MagneticStatics(abc.ABC):
+    """
+    Abstract class for `MagneticStatics` all kinds of magnetic static field
+    """
+
+    """
+    Calculate magnetic field generated by this wire at position `p`
+
+    Parameters:
+    -----------
+    p :
+        three-dimensional position vector
+
+    Returns:
+    -------
+    B : `astropy.units.Quantity`
+        magnetic field at the specified positon
+
+    """
+    @abc.abstractmethod
+    def magnetic_field(self, p) -> u.Quantity:
+        pass
+
+
+class MagneticDipole(MagneticStatics):
+    @u.quantity_input(moment=u.A*u.m*u.m, p0=u.m)
+    def __init__(self, moment: u.Quantity, p0: u.Quantity):
+        self.moment = moment.to(u.A*u.m*u.m).value
+        self.p0 = p0.to(u.m).value
+
+    def __repr__(self):
+        return ("Magnetic Dipole: \n"
+                f"  moment: {self.moment*u.m*u.m*u.A}\n"
+                f"  p0: {self.p0*u.m}\n"
+                )
```

Comment:
How does `__repr__` behave when you make it multiline? Wouldn't it be safer to go with a single line version?

Also, f-strings :+1: